vllm/sampling_params.py

```diff
@@ -33,6 +38,15 @@ class SamplingType(IntEnum):
 to sample from."""
 
 
+@dataclass
+class LogitsProcessorParams:
+    """Parameters required to construct one or more logits processors"""
+    logit_bias: Optional[Union[Dict[int, float], Dict[str, float]]]
+    allowed_token_ids: Optional[List[int]]
+    guided_decoding_request: Optional[  # type: ignore
+        'GuidedDecodingRequest']  # noqa
```

Comment:
I think these should probably just be top-level sampling params, the fact that these get turned into logits processors is a bit of an implementation detail.